File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> pMap = new HashMap<>();
        for(String p : participant) pMap.put(p, pMap.getOrDefault(p, 0) + 1);
        for(String c : completion) pMap.put(c, pMap.get(c) - 1);
        
        for(String key : pMap.keySet()){
            if(pMap.get(key) != 0) {
                answer = key;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    
    for person in participant:
        if person in dic:
            dic[person] += 1
        else:
            dic[person] = 1
    for person in completion:
        if person in dic:
            dic[person] += 1
        else:
            dic[person] = 1
    for key, val in dic.items():
        if val % 2 != 0:
            answer = key
            break
    return answer